DFC.ServiceTaxonomy.PageLocation/Validators/PageLocationsTaxonomyValidator.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DFC.ServiceTaxonomy.Content;
+using DFC.ServiceTaxonomy.Content.Services.Interface;
+using DFC.ServiceTaxonomy.Taxonomies.Helper;
+using DFC.ServiceTaxonomy.Taxonomies.Models;
+using DFC.ServiceTaxonomy.Taxonomies.Validation;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentManagement;
+
+namespace DFC.ServiceTaxonomy.PageLocation.Validators
+{
+    public class PageLocationsTaxonomyValidator : ITaxonomyValidator
+    {
+        private readonly IContentItemsService _contentItemsService;
+        private readonly ITaxonomyHelper _taxonomyHelper;
+        private IEnumerable<ITaxonomyTermValidator> _validators;
+
+        public PageLocationsTaxonomyValidator(IContentItemsService contentItemsService, ITaxonomyHelper taxonomyHelper, IEnumerable<ITaxonomyTermValidator> validators)
+        {
+            _contentItemsService = contentItemsService;
+            _taxonomyHelper = taxonomyHelper;
+            _validators = validators;
+        }
+
+        public async Task<TaxonomyValidationResult> Validate(TaxonomyPart part)
+        {
+            List<string> errors = new List<string>();
+
+            //make sure nothing has moved that has associated pages anywhere down the tree
+            List<ContentItem> allPages = await _contentItemsService.GetActive(ContentTypes.Page);
+
+            JArray? terms = _taxonomyHelper.GetAllTerms(JObject.FromObject(part.ContentItem));
+
+            if (terms != null)
+            {
+                foreach (JObject term in terms)
+                {
+                    dynamic? originalParent = _taxonomyHelper.FindParentTaxonomyTerm(term, JObject.FromObject(part.ContentItem));
+                    dynamic? newParent = _taxonomyHelper.FindParentTaxonomyTerm(term, JObject.FromObject(part));
+
+                    if (originalParent == null || newParent == null)
+                        throw new InvalidOperationException($"Could not find parent taxonomy term for {(originalParent == null ? originalParent : newParent)}");
```

Comment:
Is that ternary operator the right way round?

Reply:
It's just generally wrong, supposed to be the item it's looking for the parent for

Reply:
Fixed now!